File: conformance/bundles/05-ambiguous-match/cukes.steps.rs
//! Rust sibling of `cukes.steps.ts` (bundle `05-ambiguous-match`).

use var::{Handler, Registry, Steps, Value};

pub const FILE: &str = "cukes.steps.rs";

pub fn register(r: Registry) -> Registry {
    let mut s = Steps::from_registry(r);
    // Both expressions match "I have 5 cukes" → ambiguous-match diagnostic.
    s.stimulus(
        "I have {int} cukes",
        FILE,
        1,
        Handler::sync1(|_state, _n| Ok(None)),
    );
    s.stimulus("I have 5 cukes", FILE, 2, Handler::sync0(|_state| Ok(None)));
    s.into_registry()
}

pub fn state() -> Value {
    Value::Null
}

File: conformance/bundles/05-ambiguous-match/cukes.steps.cs
// C# sibling of cukes.steps.ts / .rs (bundle 05-ambiguous-match).
using Varar;
using Varar.Core;

namespace Varar.Corpus.B05;

public static class CukesSteps
{
    public static Registry Register(Registry r)
    {
        var s = Steps.From(r);

        // Both expressions match "I have 5 cukes" → ambiguous-match diagnostic.
        s.Stimulus("I have {int} cukes", (state, n) => null);
        s.Stimulus("I have 5 cukes", state => null);

        return s.ToRegistry();
    }

    public static Value State() => Value.Null;
}
